localstack/services/apigateway/helpers.py

```diff
@@ -108,6 +106,78 @@ def __init__(self):
         self.client_certificates = {}
 
 
+class Resolver:
+    def __init__(self, document, allow_recursive=True):
+        self.document = document
+        self.allow_recursive = allow_recursive
+        # cache which maps known refs to part of the document
+        self._cache = {}
+        self._refpaths = ["#"]
+
+    def _is_ref(self, item):
+        return isinstance(item, dict) and "$ref" in item
+
+    def _is_internal_ref(self, refpath):
```

Comment:
Would be helpful to add some type hints to the method parameters / return types in this class. 👍